mava/components/training/losses.py

```diff
@@ -332,10 +391,13 @@ def critic_loss_fn(
                     observations: Any,
                     target_values: jnp.ndarray,
                     behavior_values: jnp.ndarray,
+                    mask: jnp.ndarray,
                 ) -> Tuple[jnp.ndarray, Dict[str, jnp.ndarray]]:
                     """Inner critic loss function: see outer function for parameters."""
 
-                    values = network.critic_network.apply(critic_params, observations)
+                    values = (
+                        network.critic_network.apply(critic_params, observations) * mask
```

Comment:
Again is it necessary to mask here if we mask in the loss function?

Reply:
Great catch. This is not necessary because all the operations that follow are element wise. So we just need to mask right at the end. Same as before